Content.Shared/_Starlight/Shadekin/Componants/ShadekinComponents.cs

```diff
@@ -0,0 +1,144 @@
+using Content.Shared.Actions;
+using Content.Shared.Alert;
+using Content.Shared.DoAfter;
+using Content.Shared.FixedPoint;
+using Robust.Shared.GameStates;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization;
+
+namespace Content.Shared._Starlight.Shadekin;
+
+#region Shadekin
+[RegisterComponent, NetworkedComponent, AutoGenerateComponentPause, AutoGenerateComponentState]
+public sealed partial class ShadekinComponent : Component
+{
+    [DataField]
+    public ProtoId<AlertPrototype> ShadekinAlert = "Shadekin";
+
+    [ViewVariables(VVAccess.ReadOnly), AutoPausedField]
+    public TimeSpan NextUpdate = TimeSpan.Zero;
+
+    [DataField]
+    public TimeSpan UpdateCooldown = TimeSpan.FromSeconds(1f);
+
+    [AutoNetworkedField, ViewVariables]
+    public ShadekinState CurrentState { get; set; } = ShadekinState.Dark;
+
+    [DataField("thresholds", required: true)]
+    public SortedDictionary<FixedPoint2, ShadekinState> Thresholds = new();
+}
+
+[Serializable, NetSerializable]
+public enum ShadekinState : byte
+{
+    Invalid = 0,
+    Dark = 1,
+    Low = 2,
+    Annoying = 3,
+    High = 4,
+    Extreme = 5
+
+}
+#endregion
+
+#region OrganShadekinCoreComponent
+[RegisterComponent, NetworkedComponent]
+public sealed partial class OrganShadekinCoreComponent : Component
+{
+    [DataField]
+    public EntityUid? OrganOwner;
+
+    [DataField]
+    public bool Damaged = true;
+}
+#endregion
+
+#region Brighteye
+[RegisterComponent, NetworkedComponent, AutoGenerateComponentPause, AutoGenerateComponentState]
+public sealed partial class BrighteyeComponent : Component
+{
+    [DataField]
+    public ProtoId<AlertPrototype> BrighteyeAlert { get; set; } = "ShadekinEnergy";
+
+    [DataField]
+    public ProtoId<AlertPrototype> PortalAlert { get; set; } = "ShadekinPortalAlert";
+
+    [DataField]
+    public ProtoId<AlertPrototype> RejuvenationAlert { get; set; } = "ShadekinRejuvenateAlert";
+
+    /// <summary>
+    /// How many Energy the brighteye has.
+    /// </summary>
+    [DataField, AutoNetworkedField]
+    public int Energy = 0;
+
+    /// <summary>
+    /// The Max Energy the brighteye can have.
+    /// </summary>
+    [DataField, AutoNetworkedField]
+    public int MaxEnergy = 200;
+
+    /// <summary>
+    /// Shadekin Rejuvenating Bool
+    /// </summary>
+    [DataField]
+    public bool Rejuvenating = false;
```

Comment:
```suggestion
    [DataField, AutoNetworkedField]
    public bool Rejuvenating = false;
```
since we cant use the portal during rejuvenating

Reply:
Im gona try... but hopefully i dont have to change the poppup spamming.